### PR TITLE
APEXMALHAR-2065 #resolve #comment Added getWindowIds method to Window…

### PR DESCRIPTION
**Pull request by ilooner**
…Datamanager

### CODE REVIEW

**Comment by ilooner** [comment]
@chandnisingh please review

File: library/src/main/java/org/apache/apex/malhar/lib/wal/WindowDataManager.java [hunk 1]
```diff
@@ -227,6 +235,22 @@ public void delete(int operatorId, long windowId) throws IOException
...
+
+      int index = 0;
+
+      for (Long windowId: windowIds) {
+        windowIdsArray[index] = windowId;
+        index++;
+      }
+
+      return windowIdsArray;
+    }
```
Comment (otterc) [inline library/src/main/java/org/apache/apex/malhar/lib/wal/WindowDataManager.java:254]: Can you please document that windows ids are sorted?

Comment (ilooner) [inline library/src/main/java/org/apache/apex/malhar/lib/wal/WindowDataManager.java:254]: @chandnisingh Added to the documentation

**Comment by otterc** [comment]
@ilooner this uses `replayState` and returns window ids which are sorted, however the `getWindowIds(int operatorId)` utilizes FSStorageAgent. Can you please looking into making the older method use `replayState` as well and return a sorted array. If there are any operators that are using WindowDataManager you might have to check that we didn't break them. 

Thanks, 
Chandni

**Comment by ilooner** [comment]
@chandnisingh since the implementation of WindowDataManager will be changing soon anyway I'd like to avoid making additional changes to it.

**Comment by otterc** [comment]
@ilooner there is a check style issue at line 111. There needs to be an empty line between class/method definitions. 
Please fix it and then I will merge it.

**Comment by otterc** [comment]
@ilooner please close this. I merged it and fixed the check style.
I needed to change WindowDataManager to use Wal which is why I took the change.

Thanks,
Chandni

**Pull request closed without merging**